### PR TITLE
LPS-86824 LPS-86851 LPS-86856 fix various portlet 3 CDI issues

### PR DESCRIPTION
**Pull request by rotty3000**
cc @vicnate5 @ngriffin7a @stiemannkj1

### CODE REVIEW

**Comment by rotty3000** [comment]
ci:test:bundle

**Comment by shuyangzhou** [comment]
ci:test

**Comment by vicnate5** [comment]
Tested tickets from this pull.

PASSED https://issues.liferay.com/browse/LPS-86824 using final patch from Ray

PASSED https://issues.liferay.com/browse/LPS-86856 using patch in ticket

PASSED https://issues.liferay.com/browse/LPS-86851
Test Succeeded in all portlets

Some WARN messages
```
2019-02-12 09:30:07.925 WARN  [fileinstall-/Users/vicnate5/Liferay/cdiTests/bundles/osgi/war][Interceptor:46] WELD-001700: Interceptor annotation class javax.ejb.PostActivate not found, interception based on it is not enabled
2019-02-12 09:30:07.925 WARN  [fileinstall-/Users/vicnate5/Liferay/cdiTests/bundles/osgi/war][Interceptor:46] WELD-001700: Interceptor annotation class javax.ejb.PrePassivate not found, interception based on it is not enabled
2019-02-12 09:30:08.113 INFO  [fileinstall-/Users/vicnate5/Liferay/cdiTests/bundles/osgi/war][Event:86] WELD-000411: Observer method [BackedAnnotatedMethod] org.apache.aries.cdi.container.internal.container.DiscoveryExtension.processAnnotatedType(@Observes ProcessAnnotatedType<X>) receives events for all annotated types. Consider restricting events using @WithAnnotations or a generic type with bounds.
2019-02-12 09:30:08.586 INFO  [Aries CCR Thread][Bootstrap:225] WELD-000101: Transactional services not available. Injection of @Inject UserTransaction not available. Transactional observers will be invoked synchronously.
2019-02-12 09:30:08.622 INFO  [Aries CCR Thread][Event:86] WELD-000411: Observer method [BackedAnnotatedMethod] public com.liferay.bean.portlet.cdi.extension.internal.BeanPortletExtension.step2ProcessAnnotatedType(@Observes ProcessAnnotatedType<T>) receives events for all annotated types. Consider restricting events using @WithAnnotations or a generic type with bounds.
2019-02-12 09:30:08.756 WARN  [Aries CCR Thread][BeanPortletExtension:234] Automatically added @ApplicationScoped to class javax.portlet.tck.portlets.TestPortlet
2019-02-12 09:30:09.372 INFO  [weld-worker-2][BeanPortletExtension:428] Registered 4 bean portlets and 0 bean filters for tck-V3URLTests
```

**Comment by ngriffin7a** [comment]
@vicnate5 The warnings look appropriate to me -- works as designed.

**Comment by vicnate5** [comment]
Functional test failures not unique to this pull. but we are a bit blind because there are lots of issues at head :(

**Comment by shuyangzhou** [comment]
ci:test

**Comment by rotty3000** [comment]
ci:test

**Comment by shuyangzhou** [comment]
@rotty3000 and @ngriffin7a in all 3 tickets, no one is using "org.osgi.service.cdi.annotations" which makes the CDIAnnotations support logic not being used.

Do we have an example usage of those? Without ways to test it we won't know whether those complex logic is working or not.

In case we don't have example usage or even not planning to add it, we should defer the CDIAnnotations support until really needed, that way we will have some examples to test against.

**Comment by rotty3000** [comment]
@shuyangzhou you don't need "org.osgi.service.cdi.annotations" for `CDIAnnotations` to be used. You only need the instruction to enable CDI discovery which is the default from modules/apps/static/portal-osgi-web/portal-osgi-web-wab-generator-impl/src/main/java/com/liferay/portal/osgi/web/wab/generator/internal/processor/WabProcessor.java

**Comment by shuyangzhou** [comment]
@rotty3000 I am talking about the CDIAnnotations.java and CDIAnnotationReader.java. Most code there is no hitted on deploying those 3 wars. Due to they are very BND styled, I need to clean them up before merging. But without a way to use them, I won't know whether I break something or not.

From another angle to look at this, if we have no way to use most code in CDIAnnotations.java and CDIAnnotationReader.java, why should we carry them?

File: modules/apps/static/portal-osgi-web/portal-osgi-web-wab-generator-impl/src/main/java/com/liferay/portal/osgi/web/wab/generator/internal/cdi/CDIAnnotationReader.java [hunk 1]
```diff
@@ -0,0 +1,763 @@
...
+	}
+
+	@Override
+	public void annotation(Annotation annotation) {
+		try {
+			Descriptors.TypeRef typeRef = annotation.getName();
+
+			String fqn = typeRef.getFQN();
+
+			if ("org.osgi.service.cdi.annotations.Bean".equals(fqn)) {
```
Comment (shuyangzhou) [inline modules/apps/static/portal-osgi-web/portal-osgi-web-wab-generator-impl/src/main/java/com/liferay/portal/osgi/web/wab/generator/internal/cdi/CDIAnnotationReader.java]: @rotty3000 for example, we are not hitting any of these if conditions by deploying the testing wars. And many other branches in this class and CDIAnnotations too.

Is there a real expected use case for these?

Comment (rotty3000) [reply modules/apps/static/portal-osgi-web/portal-osgi-web-wab-generator-impl/src/main/java/com/liferay/portal/osgi/web/wab/generator/internal/cdi/CDIAnnotationReader.java]: sigh.... I'm going to take a different approach.

**Pull request closed without merging**